src/array/replace.ts

```diff
@@ -18,15 +18,13 @@ export function replace<T>(
   if (newItem === undefined) {
     return [...array]
   }
-  for (let idx = 0; idx < array.length; idx++) {
-    const item = array[idx]
-    if (match(item, idx)) {
-      return [
-        ...array.slice(0, idx),
-        newItem,
-        ...array.slice(idx + 1, array.length),
-      ]
+  const out = array.slice()
+  for (let index = 0; index < array.length; index++) {
+    const item = array[index]
+    if (match(item, index)) {
+      out.splice(index, 1, newItem)
```

Comment:
Why not just `out[index] = newItem`?  `out` is already a copy.

Reply:
Good catch!